uaclient/clouds/azure.py

```diff
@@ -49,9 +52,14 @@ def is_viable(self) -> bool:
                 return True
         return os.path.exists(AZURE_OVF_ENV_FILE)
 
-    def should_poll_for_pro_license(self) -> bool:
-        """Unsupported"""
-        return False
-
     def is_pro_license_present(self, *, wait_for_change: bool) -> bool:
-        raise exceptions.InPlaceUpgradeNotSupportedError()
+        url = IMDS_URLS.get("compute", "")
+        try:
+            data, headers = util.readurl(url, headers={"Metadata": "true"})
+        except HTTPError as e:
+            LOG.error(e)
+            if e.code == 400:
+                raise exceptions.CancelProLicensePolling()
+            else:
+                raise exceptions.DelayProLicensePolling()
```

Comment:
This is probably fine, but because there is no long polling these two cases have the same effect, so we could just do:
```suggestion
            LOG.error(e)
            raise exceptions.CancelProLicensePolling()
```